actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/tenantpreference/TenantPreferenceManagementActor.java

```diff
@@ -64,86 +64,94 @@ public void onReceive(Request request) throws Throwable {
+    properties.put(JsonKey.ORG_ID, orgId);
+    Response tenantPreferences =
+        cassandraOperation.getRecordsByProperties(
+            tenantPreferenceDbInfo.getKeySpace(),
+            tenantPreferenceDbInfo.getTableName(),
+            properties);
+    List<Map<String, Object>> preferencesList =
+        (List<Map<String, Object>>) tenantPreferences.get(JsonKey.RESPONSE);
+    return preferencesList;
+  }
   /**
    * Method to update the Tenant preference on basis of id or (role and org id).
    *
    * @param actorMessage
    */
-  @SuppressWarnings("unchecked")
   private void updateTenantPreference(Request actorMessage) {
 
-    String orgId = (String) actorMessage.getRequest().get(JsonKey.ROOT_ORG_ID);
+    Map<String, Object> req = actorMessage.getRequest();
+    String orgId = (String) req.get(JsonKey.ORG_ID);
+    String key = (String) req.get(JsonKey.KEY);
     ProjectLogger.log(
-        "TenantPreferenceManagementActor-updateTenantPreference called for org: " + orgId);
-    validateRequest(actorMessage, true);
-
+        "TenantPreferenceManagementActor-updateTenantPreference called for org: "
+            + orgId
+            + "key "
+            + key);
     Response finalResponse = new Response();
-    List<Map<String, Object>> responseList = new ArrayList<>();
-    List<Map<String, Object>> reqList =
-        (List<Map<String, Object>>) actorMessage.getRequest().get(JsonKey.TENANT_PREFERENCE);
-    List<Map<String, Object>> preferencesList = getPreferencesFromDB(orgId);
-    for (Map<String, Object> map : reqList) {
-      Map<String, Object> preferenceObj = null;
-      String key = (String) map.get(JsonKey.KEY);
-      String data = (String) map.get(JsonKey.DATA);
-      // skip update if either key or data is empty
-      if (StringUtils.isNotBlank(key) && StringUtils.isNotBlank(data)) {
-        boolean found = false;
-        for (Map<String, Object> m : preferencesList) {
-          // check if preference key exists for the org
-          if (key.equals(m.get(JsonKey.KEY))) {
-            preferenceObj = m;
-            found = true;
-            break;
-          }
-        }
-        // if preference is not found
-        if (!found)
-          responseList.add(
-              getResponseMap(
-                  orgId,
-                  key,
-                  "Preference setting not found for key: " + key + " for the org: " + orgId));
-        // if preference is found
-        if (null != preferenceObj) {
-          preferenceObj.put(JsonKey.KEY, key);
-          preferenceObj.put(JsonKey.DATA, data);
-          cassandraOperation.updateRecord(
-              tenantPreferenceDbInfo.getKeySpace(),
-              tenantPreferenceDbInfo.getTableName(),
-              preferenceObj);
-          responseList.add(getResponseMap(orgId, key, null));
-        }
+    List<Map<String, Object>> preferencesList = getTenantPreferencesFromDB(orgId, key);
+    if (CollectionUtils.isEmpty(preferencesList)) {
```

Comment:
Recall we may like to reset to empty - to indicate no preferences. Should this exception be needed. I think we can allow empty map.
Reading this am thinking we must have a delete API as well - just to remove this in total.

Reply:
@indrajra this preference we are reading from DB, based on key and orgId in the update req, to update the data the preference should be present in DB, if not we have to throw not found exception right?

Reply:
if they need to update empty map data, then I should not check emptiness, that I am doing in TenantPreferenceValidator

Reply:
ok.